modules/core/src/main/scala/com.fortysevendeg.hood/csv/CsvService.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2019 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.fortysevendeg.hood.csv
+
+import java.io.File
+
+import cats.data.EitherT
+import cats.effect.Sync
+import com.fortysevendeg.hood.model._
+import kantan.csv._
+import kantan.csv.ops._
+import kantan.csv.generic._
+import cats.implicits._
+import io.chrisdavenport.log4cats.Logger
+
+import scala.io.Source
+
+trait CsvService[F[_]] {
+
+  def parseBenchmark(csvPath: File): F[Either[HoodError, List[Benchmark]]]
+
+}
```

Comment:
if we are going to use actual files to test this I think we don't need tagless final here, isn't it?

Reply:
`io` operations are always side-effects that we want to deal with. I'm in favor of F-algebra here

Reply:
I agreed but I was just thinking we don't need the trait and can keep the object or transform it into a class if necessary

Reply:
@AdrianRaFo @juanpedromoreno I can work either way. The requirements specified an algebra for this, that's why I followed that approach (with the intention of combining everything when building the actual plugin). What are your thoughts on this? Should I keep it as an actual algebra in this case?